src/components/Api.js

```diff
@@ -0,0 +1,153 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+    // constructor body
+  }
+
+  async getInitialCards() {
+    try {
+      const res = await fetch(`${this._baseUrl}/cards`, {
+        method: "GET",
+        headers: this._headers,
+      });
+      if (!res.ok) {
+        throw new Error(`HTTP error! Status: ${res.status}`);
+      }
+      return await res.json();
```

Comment:
```js
.then(res => {
                if (res.ok) {
                    return res.json();
                }
                return Promise.reject(`Error ${res.status}`);
            })

```

The code of `ok` checking is repeated in **every** request of the project. To avoid it you need to make a special function `checkResponse` that will check any response from the server if it’s ok or not.


```js
function checkResponse(res) {
        // here is the code of the checking
    }

```

And now you can replace redundant repeating with the only line:


```js
 .then(checkResponse)

```

Please, pay attention that you need to pass only the **reference** for the function rather than the call. 

----------------------------------------------------------------

**Could be improved**

You can make a special function for fetching and checking responses **not to duplicate** it in every request:

```js

  function request(url, options) {
    return fetch(url, options).then(checkResponse)
  }

```

And now you can safely replace `fetch` with `request` and remove repeated checking. And you don’t need to change anything else.

Reply:
i completed this, however in my comments it says i didnt. is there something else that i am missing. please let me know.